src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -62,11 +64,47 @@ private static Action<ModuleSymbol> ValidateRequiredMembersInModule(string[] mem
                 {
                     AssertEx.Any(member.GetAttributes(), attr => attr.AttributeClass.ToTestDisplayString() == "System.Runtime.CompilerServices.RequiredMemberAttribute");
                 }
-                Assert.True(((NamedTypeSymbol)member.ContainingSymbol).HasDeclaredRequiredMembers);
+
+                requiredTypes.Add((NamedTypeSymbol)member.ContainingType);
+            }
+
+            foreach (var type in requiredTypes)
+            {
+                AssertTypeRequiredMembersInvariants(module, type);
             }
         };
     }
 
+    private static Action<ModuleSymbol> AssertTypeRequiredMembersInvariants(string expectedType)
```

Comment:
nit: this method doesn't assert. Consider `GetTypeRequiredMembersInvariantsValidator`